### ISSUE TITLE
Using `source deactivate` on a conda env fails unless the root anaconda directory is in PATH

### ISSUE THREAD

**Issue by goxberry** [conda/conda#458]
tl;dr: Is the deactivate script meant to support a setup in which `CONDA_DEFAULT_ENV` is not added to `PATH` via a configuration file, but rather added via `source $CONDA_DEFAULT_ENV/bin/activate $CONDA_DEFAULT_ENV`?

Possibly relevant: #420, #339, and #116.

Not sure if this is really a bug, but here's a close to minimal reproducible example using Anaconda 1.8.0 on Mac OS X 10.8.5:

```
Geoffs-MacBook-Pro:anaconda goxberry$ source ~/anaconda/bin/activate ~/anaconda
prepending /Users/goxberry/anaconda/bin to PATH
(/Users/goxberry/anaconda)Geoffs-MacBook-Pro:anaconda goxberry$ conda create -p env1 anaconda

Package plan for creating environment at /Users/goxberry/anaconda/env1:
```

(snip)

```
Linking packages ...
ln: /Users/goxberry/Desktop//Launcher.app: File exists###############################################        |  35%
[      COMPLETE      ] |###############################################################| 100%
#
# To activate this environment, use:
# $ source activate /Users/goxberry/anaconda/env1
#
# To deactivate this environment, use:
# $ source deactivate
#
(/Users/goxberry/anaconda)Geoffs-MacBook-Pro:anaconda goxberry$ source activate /Users/goxberry/anaconda/env1/
discarding /Users/goxberry/anaconda/bin from PATH
prepending /Users/goxberry/anaconda/env1/bin to PATH
(/Users/goxberry/anaconda/env1/)Geoffs-MacBook-Pro:anaconda goxberry$ source deactivate
-bash: deactivate: No such file or directory
```

Here's where I don't understand the message returned: `deactivate` isn't on my `PATH` because I don't put anaconda on my `PATH` by default. I use `source ~/anaconda/bin/activate ~/anaconda` for that, and when `conda` creates envs, there is no `deactivate` script in `~/anaconda/env1/bin`. So when I want to deactivate the env using `source deactivate`, as directed, it fails, predictably.

In order for the documentation to hold, this script must be on the `PATH`, so in theory, symlinking the script from `CONDA_DEFAULT_ENV` into new envs that are created would resolve this problem, as shown below:

```
(/Users/goxberry/anaconda/env1/)Geoffs-MacBook-Pro:anaconda goxberry$ echo $PATH
/Users/goxberry/anaconda/env1/bin:/Users/goxberry/.cabal/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
(/Users/goxberry/anaconda/env1/)Geoffs-MacBook-Pro:anaconda goxberry$ source ~/anaconda/bin/deactivate
discarding /Users/goxberry/anaconda/env1/bin from PATH
```

I grepped through the Python & Bash source to see where I might add a command to accomplish the symlinking, but soon realized it would take a while. If someone could point me to where I should focus my attention, I'd be happy to submit a pull request; otherwise, it may take me a while to figure it out.
